doc/introduction/remote.rst

```diff
@@ -0,0 +1,511 @@
+
+
+**Note:** You may use :func:`~.random_interferometer` to generate arbitrary random unitaries.
+
+This program will execute following the same steps as above; :class:`~.RemoteEngine` will automatically
+compile the program to match the layout of the chip.
+
+You may wish to view the compiled program; this can be easily done in Python using
+the ``Program.compile`` method:
+
+
+>>> from strawberryfields import RemoteEngine
+>>> from strawberryfields.io import load
+>>> prog = load("test.xbb")
+>>> prog = prog.compile("chip2")
+>>> prog.print()
+S2gate(1, 0) | (q[0], q[4])
+S2gate(1, 0) | (q[3], q[7])
+S2gate(1, 0) | (q[2], q[6])
+MZgate(1.573, 4.368) | (q[2], q[3])
+MZgate(1.573, 4.368) | (q[6], q[7])
+S2gate(1, 0) | (q[1], q[5])
+MZgate(1.228, 5.006) | (q[0], q[1])
+MZgate(4.414, 3.859) | (q[1], q[2])
+MZgate(2.98, 3.316) | (q[2], q[3])
+Rgate(-0.7501) | (q[3])
+MZgate(5.397, 5.494) | (q[0], q[1])
+MZgate(5.152, 4.891) | (q[1], q[2])
+Rgate(2.544) | (q[2])
+MZgate(1.228, 5.006) | (q[4], q[5])
+MZgate(4.414, 3.859) | (q[5], q[6])
+MZgate(2.98, 3.316) | (q[6], q[7])
+Rgate(-0.7501) | (q[7])
+MZgate(5.397, 5.494) | (q[4], q[5])
+MZgate(5.152, 4.891) | (q[5], q[6])
+Rgate(2.544) | (q[6])
+Rgate(-1.173) | (q[1])
+Rgate(1.902) | (q[4])
+Rgate(1.902) | (q[0])
+Rgate(-1.173) | (q[5])
+MeasureFock | (q[0], q[1], q[2], q[3], q[4], q[5], q[6], q[7])
+
+and even saved as a new Blackbird script using the :func:`~.io.save` function:
+
+>>> from strawberryfields.io import save
+>>> save("test_compiled.xbb", prog)
+
+
+Tips and tricks
+---------------
+
+.. code-block:: python3
+
+    from strawberryfields.utils import operation
+
+We can define an operation to make it easier to apply the same unitary
+to both signal and idler modes.
+
+.. code-block:: python3
+
+    @operation(4)
+    def unitary(q):
+        ops.Interferometer(U) | q
+        ops.BSgate(0.543, 0.123) | (q[2], q[0])
+
+    prog = sf.Program(8)
+
+    with prog.context as q:
+        ops.S2gate(1.0) | (q[0], q[4])
+        ops.S2gate(1.0) | (q[1], q[5])
+        ops.S2gate(1.0) | (q[2], q[6])
+        ops.S2gate(1.0) | (q[3], q[7])
+
+        unitary() | q[:4]
+        unitary() | q[4:]
+
+
+Embedding bipartite graphs
+--------------------------
+
+We can embed bipartite graphs, with the restriction that the singular
+values form the set :math:`\{0, d\}` for some real value :math:`d`.
+
+The matrix :math:`B` represents the edges between the two sets of
+vertices in the graph, and :math:`A` is the full adjacency matrix
+:math:`A = \begin{bmatrix}0 & B\\ B^T & 0\end{bmatrix}`. Here, we will
+consider a complete bipartite graph, since we know that the singular
+values are of the form :math:`\{d, 0\}`.
+
+.. code-block:: python3
+
+    B = np.ones([4, 4])
+    A = np.block([[0*B, B], [B.T, 0*B]])
+
+    prog = sf.Program(8)
+
+    # the following mean photon number per mode
+    # quantity is set to ensure that the singular values
+    # are scaled such that all squeezers have value 1
```

Comment:
```suggestion
    # are scaled such that all Sgates have value squeezing value r=1
```

Reply:
Can we restructure this section so that it foremost shows users how to rescale their adjacency matrix (if possible) to give a runnable circuit?

Also, is the value `r=1` an actual physical constraint? I thought it was just binary :thinking:

Reply:
Not sure if this was committed, but there was a typo in my suggestion